### PR TITLE
Test master

### PR DESCRIPTION
**Pull request by GiuseppeFasanella**
Important changes:
fix in run ranges runF
fix in eos path in energyScale.sh
eleID 2016 loose WP added in SimpleCutBasedElectronIDSelectionFunctor.h

eleID to-do list:
1. deta has to be re-implemented/checked with the official recipe (?)
2. pfIso: check the effective areas (it's correct if 8_0_X is defined)
3. maxNumberOfExpectedMissingHits: it's correct if 7_2_X is defined (check)

### CODE REVIEW

File: ZFitter/script/functions.sh [hunk 1]
```diff
@@ -51,52 +52,25 @@ mcName(){
...
 mkSmearerCatSignal(){
-    #if [ ! -e "data/smearerCat/smearerCat_`basename $1 .dat`_s1-`basename $configFile .dat`.root" -o  ! -e "data/smearerCat/smearerCat_`basename $1 .dat`_s2-`basename $configFile .dat`.root" -o  ! -e "data/smearerCat/smearerCat_`basename $1 .dat`_s3-`basename $configFile .dat`.root" ];then
     basenameConfig=`basename $1 .dat`
-    if [ ! -e "data/smearerCat/smearerCat_`basename $1 .dat`_s1-`basename $configFile .dat`.root" ];then
+    if [ ! -e "${eos_path}/data/smearerCat/smearerCat_`basename $1 .dat`_s1-`basename $configFile .dat`.root" ];then
 	echo "[STATUS] Creating smearerCat for signal: `basename $configFile .dat` `basename $1 .dat`"
+	echo "./bin/ZFitter.exe -f ${configFile} --regionsFile=$1 --saveRootMacro  --addBranch=smearerCat_s"
  	./bin/ZFitter.exe -f ${configFile} --regionsFile=$1  \
- 	    --saveRootMacro  --addBranch=smearerCat_s  || exit 1
+ 	    --saveRootMacro  --addBranch=smearerCat_s #|| exit 1 -->dummy seg faults forced me to comment this exit 1 -->FIX ME
```
Comment (previsualconsent) [inline ZFitter/script/functions.sh]: Any ideas about the source of segfaults? I will try to find it.

File: ZFitter/script/functions.sh [hunk 2]
```diff
@@ -110,37 +84,16 @@ mkSmearerCatData(){
...
-#    echo "file of mkSmearerCatData is" $3
-#    cat $3.tmp >> $3
-#    rm $3.tmp $3.tmp2
-
-##adapted from mkSmearerSignal -->This works
-#Once cat files exist, just write them in the bare configfile
+
+    #Once cat files exist, just write them in the bare configfile
+    #TO-DO -> be sure that the writing is unique to avoid stupid crashes
     for tag in `grep "^d" $3 | grep selected | awk -F" " ' { print $1 } '`
```
Comment (previsualconsent) [inline ZFitter/script/functions.sh]: Re read the context, and I think I understand. Writing a fix now.

File: ZFitter/script/Init_calibration_procedure.sh [hunk 3]
```diff
@@ -11,15 +11,15 @@ fi
 #add branch r9 => do not use it until the r9 corrections are available
 ##./script/addBranch.sh data/validation/${file}.dat R9Eleprime
 #make pileupHist: they are used to make pileupTrees, and also they are required for step1
-pileupHist
+##pileupHist
```
Comment (GiuseppeFasanella) [inline ZFitter/script/Init_calibration_procedure.sh]: True, probably it's worth adding a check "if the histogram already exists, skips this part" and a "force statement" via command line

File: ZFitter/script/bash_functions_calibration.sh [hunk 4]
```diff
@@ -20,7 +20,8 @@ pileupTrees(){
 for tag in `grep "^s" data/validation/${file}.dat | grep selected | awk -F" " ' { print $1 } '`
 do
     #tag is s1, s2, ...
-    mv tmp/mcPUtree${tag}.root  ${eos_path}/data/puTree/${tag}_${file}.root
+    mv tmp/mcPUtree${tag}.root ${eos_path}/data/puTree/${tag}_${file}.root
+    #mv tmp/${tag}_pileup_chain.root  ${eos_path}/data/puTree/${tag}_${file}.root #this does not work
```
Comment (GiuseppeFasanella) [inline ZFitter/script/bash_functions_calibration.sh]: Ok. At the moment it's there to remind me not to use tmp/${tag}_pileup_chain.root

File: ZFitter/script/functions.sh (hunk 2 above)
Comment (GiuseppeFasanella) [inline ZFitter/script/functions.sh]: Ok, thanks. The fix here would be "if the category file is already written in the validation file, do not write it again".

**Comment by GiuseppeFasanella** [comment]
Probably, at the moment the most important fix is to be sure that the dummy seg faults we're having do not cause skipping the following commands in the script. For example, the categorization files are correctly produced and they go in the tmp/ but sometimes they are not moved where they are supposed to be moved and/or not written in the validation file. As a results, all the jobs fail, even if we have all the ingredients in place. The "move" and "write" by hand should be avoided

**Comment by GiuseppeFasanella** [comment]
For the categorization improvement, I can propose something like that: 
https://github.com/ECALELFS/ECALELF/blob/workflow_test/ZFitter/script/functions.sh#L56

and to be sure not to write twice the same categorization files, I propose having a variable like this:
https://github.com/ECALELFS/ECALELF/blob/workflow_test/ZFitter/script/functions.sh#L72

**Comment by previsualconsent** [comment]
Close with  	c34c185

**Pull request closed without merging**